utils/eventListeners.ts

```diff
@@ -201,3 +201,130 @@ export const waitForRewards = async (
       }
     });
   });
+
+type CodecOrArray = Codec | Codec[];
+
+const processCodecOrArray = (codec: CodecOrArray, fn: (c: Codec) => any) =>
+  Array.isArray(codec) ? codec.map(fn) : fn(codec);
+
+const toHuman = (codec: CodecOrArray) =>
+  processCodecOrArray(codec, (c) => c?.toHuman?.() ?? c);
+const toJson = (codec: CodecOrArray) =>
+  processCodecOrArray(codec, (c) => c?.toJSON?.() ?? c);
+const toHex = (codec: CodecOrArray) =>
+  processCodecOrArray(codec, (c) => c?.toHex?.() ?? c);
+
+export const matchSnapshot = (
+  codec: CodecOrArray | Promise<CodecOrArray>,
+  message?: string
+) => {
+  return expect(Promise.resolve(codec).then(toHuman)).resolves.toMatchSnapshot(
+    message
+  );
+};
+
+export const expectEvent = (codec: CodecOrArray, event: any) => {
+  return expect(toHuman(codec)).toEqual(
+    expect.arrayContaining([expect.objectContaining(event)])
+  );
+};
+
+export const expectHuman = (codec: CodecOrArray) => {
+  return expect(toHuman(codec));
+};
+
+export const expectJson = (codec: CodecOrArray) => {
+  return expect(toJson(codec));
+};
+
+export const expectHex = (codec: CodecOrArray) => {
+  return expect(toHex(codec));
+};
```

Comment:
those expects are test expectations, perhaps we should move them to the utils/setupJest.ts expects